### PR TITLE
Replace boost::unique_lock with std::unique_lock

### PR DESCRIPTION
**Pull request by JoeLoser**
Came up in a discussion from https://github.com/ripple/rippled/pull/2613 

Reviewer:
@wilsonianb

### CODE REVIEW

**Comment by nbougalis** [comment]
Are they available in MSVC? Also do we really need the `timed` variants?

**Comment by HowardHinnant** [comment]
I'm not sure what version of VS they came into, but I do know that Microsoft was extremely active in this area in the run-up to C++14.  So it is likely that VS-2015 has these facilities.

We don't need the `timed` variants, but the untimed variants are not available until C++17.  This was Microsoft's doing.  I proposed `shared_mutex` and `upgrade_mutex` for C++14.  Both implicitly had timed waits.  `upgrade_mutex` got dropped and `shared_mutex` got renamed to `shared_timed_mutex`, and then `shared_mutex` (without timed waits) got added to C++17.

**Comment by JoeLoser** [comment]
@HowardHinnant I didn't see any uses of `shared_time_mutex`, but I did find some `boost::shared_lock` instances. So I just replace those with their `std`-equivalent here in b00b912c52a44c4bf5743a4b773ab87a7a08e4fa

**Comment by HowardHinnant** [comment]
Sorry, I wasn't clear.  I'm recommending we eliminate the dependence on boost completely by replacing `boost::shared_mutex` with `std::shared_timed_mutex`.  They are roughly equivalent.  "shared_mutex" was renamed to "shared_timed_mutex" during the standardization process.

File: src/ripple/app/misc/ValidatorList.h [hunk 1]
```diff
@@ -29,9 +29,8 @@
 #include <ripple/protocol/PublicKey.h>
 #include <boost/iterator/counting_iterator.hpp>
 #include <boost/range/adaptors.hpp>
-#include <boost/thread/locks.hpp>
-#include <boost/thread/shared_mutex.hpp>
 #include <mutex>
```
Comment (wilsonianb) [inline src/ripple/app/misc/ValidatorList.h]: I think `<mutex>` can be removed

Comment (wilsonianb) [reply src/ripple/app/misc/ValidatorList.h]: Actually, I think it wasn't needed before these changes. So now removal counts as bonus points.

**Comment by nbougalis** [comment]
Merged with #2629 as 79d819584f5d92d569e4b32524b968b26200bbf8. Thanks for your contribution!

**Pull request closed without merging**